functorch/_src/eager_compilation.py

```diff
@@ -182,3 +182,16 @@ def backward(ctx, *args):
 
 def tvm_function(fn, name):
     return compiled_function(fn, partial(tvm_compile, name=f'fw_{name}'), partial(tvm_compile, name=f'bw_{name}'))
+
+def compiled_module(mod, fw_compiler, bw_compiler):
+    fmod, fparams = make_functional(mod)
+    compiled_f = compiled_function(fmod, fw_compiler, bw_compiler)
+
+    class CompiledModule(nn.Module):
+        def __init__(self):
+            super(CompiledModule, self).__init__()
+            self.orig_module = mod
+
+        def forward(self, *args, **kwargs):
+            return compiled_f(fparams, *args, **kwargs)
```

Comment:
I'm a little bit worried that this isn't going to work properly.

I suspect we probably need to pull off the params on every invocation.

Reply:
Hm, really?  What do you think would go wrong versus this approach?  And does "pulling off the params" mean we would need to do a bunch of the work in `make_functional` every time we call a compiled module?

Reply:
Imagine

```
module = compiled_module(module)
module(args)
module.weight = <new param>
module(args)
```

I think this approach would continue to hold a reference to the old param of `module.weight`.

Reply:
Okay yeah this approach is definitely broken, and I wasn't testing it thoroughly enough.  I think I don't understand the reference vs value semantics at work here; I think the "pulling off of params" done by make_functional must be giving me a new value, because when I update the params (e.g., `mod.param -= mod.param.grad`) it doesn't get reflected in `fparams`.

I guess I have to re-extract the params on every forward, like you suggested?